sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java

```diff
@@ -396,7 +396,7 @@ public void putFloats(int rowId, int count, byte[] src, int srcIndex) {
       Platform.copyMemory(src, Platform.BYTE_ARRAY_OFFSET + srcIndex, floatData,
           Platform.DOUBLE_ARRAY_OFFSET + rowId * 4L, count * 4L);
     } else {
-      ByteBuffer bb = ByteBuffer.wrap(src).order(ByteOrder.LITTLE_ENDIAN);
+      ByteBuffer bb = ByteBuffer.wrap(src).order(ByteOrder.BIG_ENDIAN);
```

Comment:
Could you please let us know which test cases are failed due to this code on `s390x`?

Reply:
This change looks risky. Any reason?

Reply:
I agree. Once we know the test cases that cause failures, I will double-check by running them on another big endian environment.

Reply:
below are the test suites which had some test failing initially and then passed on doing the above source code changes

spark/sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/compression/PassThroughEncodingSuite.scala

spark/sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnVectorSuite.scala

spark/sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/InMemoryColumnarQuerySuite.scala

spark/sql/core/src/test/scala/org/apache/spark/sql/DataFrameTungstenSuite.scala

Reply:
Hm, I guess the changes make sense? we don't test the big-endian branch on Jenkins, so I'm not surprised it still passes Jenkins. For a big-endian platform, indeed, wouldn't you have to read floats from the ByteBuffer as big-endian?

Reply:
Thanks, I will test them on ppc64 linux (BE).